src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseBaseKernels.hpp

```diff
@@ -1463,20 +1467,17 @@ class ScalingForSystemSolutionKernelFactory
                    real64 const maxAbsolutePresChange,
                    real64 const maxCompFracChange,
                    real64 const maxRelativeCompDensChange,
+                   arrayView1d< real64 const > const pressure,
+                   arrayView2d< real64 const, compflow::USD_COMP > const compDens,
+                   arrayView1d< real64 > pressureScalingFactor,
+                   arrayView1d< real64 > compDensScalingFactor,
                    globalIndex const rankOffset,
                    integer const numComp,
                    string const dofKey,
                    ElementSubRegionBase & subRegion,
                    arrayView1d< real64 const > const localSolution )
   {
-    arrayView1d< real64 const > const pressure =
```

Comment:
why do you prefer this way? passing as arguments to `createAndLaunch` instead of grabbing inside?

Reply:
I am also confused by this.

Reply:
so the code can be reused for res and wells.   getFields  is scoped with well and reservoir uses flow ...plus IMHO the prop id shouldn't be bundle with its shape

Reply:
this kind of call ` subRegion.getField< fields::flow::globalCompDensity >(); ` should be totally valid also in the case of a well.

Reply:
oh I see, we have separate traits for the wells. Unless we are doing it so that the dofManager uses different labels we have no reason to do so.

Reply:
yes the could.. but they haven't been....   the DECLARE_FIELD has the array shape in it along with other fields.. Im familiar with having an independent  property GUID system that can be used  in subsystems for allocation    , defining what is serialized , units for property , etc..  but not all bundled together...

Reply:
well, that macro has everything need to build a trait to register a wrapper on a group. At the end of the day, we need to to know the types of the objects... we could certalize the definition of the keys to a single location, or one per physicsPackage.

Reply:
"we could certalize the definition of the keys to a single location, or one per physicsPackage".   I support that....